tools/stress-cluster/scripts/deploy_stress_tests.ps1

```diff
@@ -0,0 +1,134 @@
+[CmdletBinding(DefaultParameterSetName = 'Default')]
+param(
+    [string]$searchDirectory,
+    [hashtable]$filters,
+    [string]$environment,
+    [string]$uploadToRegistry,
+    [string]$repository,
+
+    [Parameter(Mandatory=$true)]
+    [string]$deployId,
+
+    [Parameter(ParameterSetName = 'DoLogin', Mandatory = $true)]
+    [string]$subscription,
+
+    [Parameter(ParameterSetName = 'DoLogin', Mandatory = $true)]
+    [string]$clusterName,
+
+    [Parameter(ParameterSetName = 'DoLogin', Mandatory = $true)]
+    [string]$clusterGroup
+)
+
+. $PSScriptRoot/find_all_stress_packages.ps1
+
+# Powershell does not (at time of writing) treat exit codes from external binaries
+# as cause for stopping execution, so do this via a wrapper function.
+# See https://github.com/PowerShell/PowerShell-RFC/pull/277
+function run() {
+    Write-Output "" "==> $args" ""
+    $command, $arguments = $args
+    & $command $arguments
+    if (!$?) {
+        $code = $LASTEXITCODE
+        Write-Output "Command '$args' failed with code: $code"
+        exit $code
```

Comment:
Do you really want the first failed run to exit? Will there be any necessary clean-up that might need to happen?

Reply:
Ack, good catch...side effect of removing jobs which I was relying on for this behavior as well.